packages/rest/src/resource.ts

```diff
@@ -143,6 +143,29 @@ export function $parentResource({ program }: DecoratorContext, entity: Type, par
   }
 
   program.stateMap(parentResourceTypesKey).set(entity, parentType);
+
+  // Ensure that the parent resource type(s) don't have key name conflicts
+  const keyNameSet = new Set<string>();
+  let currentType: ModelType | undefined = entity as ModelType;
+  while (currentType) {
+    const resourceKey = getResourceTypeKey(program, currentType);
+    const keyName = getKeyName(program, resourceKey.keyProperty);
+    if (keyNameSet.has(keyName)) {
+      reportDiagnostic(program, {
+        code: "duplicate-parent-key",
+        format: {
+          resourceName: (entity as ModelType).name,
+          parentName: currentType.name,
+          keyName,
+        },
+        target: entity,
```

Comment:
should the target maybe be the 2 properties with duplicates(emit 2 diagnostic like it does for duplicate operation)

Reply:
Since the child type is the one that would technically be the "offender", I think it'd be sufficient to mark only the property of the child type.   

However, you did help me realize that I should be applying the diagnostic to the property with the `@key` decorator instead of the resource type!